src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13432,7 +13432,12 @@ bool shouldParseSeparatorOrElement()
                 // We're not on a valid separator, but we want to be resilient for the user accidentally using the wrong
                 // one in common cases.
                 if (allowSemicolonAsSeparator && this.CurrentToken.Kind is SyntaxKind.SemicolonToken)
```

Comment:
i'm a little confused by this.  why not just say allowSemicolonAsSeparator  for these initializers.

Reply:
?

Reply:
Because, as I last mentioned in the issue, doing this alone makes another similar case worse and your suggestion was to only eat the token as a comma if what follows after can continue.

Here: https://github.com/dotnet/roslyn/issues/71065#issuecomment-1898904826

Reply:
Sorry for the confusion, you were correct this logic wasn't doing anything.